File: Reto # - 5/Python/PorValor.py
## DeathbatO

def cambiar_valor(a, b):
    temp = a
    a = b
    b = temp

    return a, b

a_original = 24
b_original = 56

nuevo_a, nuevo_b = cambiar_valor(a_original, b_original)

print("Por valor:")
print("Original: a =", a_original, "b =", b_original)
print("Nuevos: a =", nuevo_a, "b =", nuevo_b)

File: Reto # - 5/C++/PorValor.cpp
//DeathbatO

#include <iostream>

// Función para intercambiar valores por valor
void intercambiarValor(int a, int b, int &nuevoA, int &nuevoB) {
    // Intercambiar valores
    int temp = a;
    nuevoA = b;
    nuevoB = temp;
}

int main() {
    // Definir variables
    int originalA = 5;
    int originalB = 10;

    // Llamar a la función e imprimir resultados
    int nuevoA, nuevoB;
    intercambiarValor(originalA, originalB, nuevoA, nuevoB);

    std::cout << "Por valor:" << std::endl;
    std::cout << "Original: a = " << originalA << ", b = " << originalB << std::endl;
    std::cout << "Nuevos:   a = " << nuevoA << ", b = " << nuevoB << std::endl;

    return 0;
}